src/coreclr/src/jit/importer.cpp

```diff
@@ -3863,11 +3863,13 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
             // to the call to the first slot of the ByReference struct.
             op1                                    = impPopStack().val;
             GenTree*             thisptr           = newobjThis;
+            GenTreeLclVar*       lclVar            = thisptr->gtGetOp1()->AsLclVar();
             CORINFO_FIELD_HANDLE fldHnd            = info.compCompHnd->getFieldInClass(clsHnd, 0);
             GenTree*             field             = gtNewFieldRef(TYP_BYREF, fldHnd, thisptr, 0);
             GenTree*             assign            = gtNewAssignNode(field, op1);
-            GenTree*             byReferenceStruct = gtCloneExpr(thisptr->gtGetOp1());
-            assert(byReferenceStruct != nullptr);
+            GenTreeLclVar*       byReferenceStruct = gtCloneExpr(lclVar)->AsLclVar();
+            assert(!byReferenceStruct->CanCSE());
```

Comment:
Did you want to remove the assert?  I believe that gtCloneExpr can return nullptr for some complex trees.
`assert(byReferenceStruct != nullptr);`

Reply:
`CanCSE` does `this` dereferencing, so it is not necessary to check it separately, but I will return it if it reads better.

Reply:
An A/V in a Checked build is worse than an assert firing, in terms of debuggablity.